### PR TITLE
[Model][Qwen3VL] Simplify `get_mrope_input_positions` using numpy

### PR DESCRIPTION
**Pull request by lgeiger**
## Purpose

This PR simplifies the Qwen3VL `get_mrope_input_positions` computation by using `np.indices` to make the code more readable.

In a profile I also noticed that the torch CPUs ops, especially `torch.tensor(list[int])` are slower than their numpy equivalents so this PR also changes the computation to numpy.

**Before:**

Screenshot 2025-11-07 at 15 14 40
Screenshot 2025-11-07 at 15 15 10

**After:**

Screenshot 2025-11-07 at 15 15 24
Screenshot 2025-11-07 at 15 14 53

## Test Plan

```
VLLM_WORKER_MULTIPROC_METHOD=spawn lm_eval --model vllm-vlm --model_args "pretrained=Qwen/Qwen3-VL-30B-A3B-Instruct-FP8,max_model_len=10000" --tasks chartqa --batch_size auto --apply_chat_template
```

## Test Result

**Before:**

| Tasks   | Version | Filter | n-shot | Metric            |     |  Value |     | Stderr |
| ------- | ------: | ------ | -----: | ----------------- | --- | -----: | --- | -----: |
| chartqa |       0 | none   |      0 | anywhere_accuracy | ↑   | 0.8680 | ±   | 0.0068 |
|         |         | none   |      0 | exact_match       | ↑   | 0.6340 | ±   | 0.0096 |
|         |         | none   |      0 | relaxed_accuracy  | ↑   | 0.8572 | ±   | 0.0070 |

**After:**

| Tasks   | Version | Filter | n-shot | Metric            |     |  Value |     | Stderr |
| ------- | ------: | ------ | -----: | ----------------- | --- | -----: | --- | -----: |
| chartqa |       0 | none   |      0 | anywhere_accuracy | ↑   | 0.8672 | ±   | 0.0068 |
|         |         | none   |      0 | exact_match       | ↑   | 0.6324 | ±   | 0.0096 |
|         |         | none   |      0 | relaxed_accuracy  | ↑   | 0.8576 | ±   | 0.0070 |

### CODE REVIEW

File: vllm/model_executor/models/qwen3_vl.py [hunk 1]
```diff
@@ -1484,43 +1482,24 @@ def get_mrope_input_positions(
...
-                torch.arange(text_len).view(1, -1).expand(3, -1) + st_idx
+                np.broadcast_to(np.arange(text_len), (3, text_len)) + st_idx
             )
 
-        llm_positions = torch.cat(llm_pos_ids_list, dim=1).reshape(3, -1)
+        llm_positions = np.concatenate(llm_pos_ids_list, axis=1).reshape(3, -1)
         mrope_position_delta = (llm_positions.max() + 1 - len(input_tokens)).item()
         llm_positions = llm_positions[:, context_len:seq_len]
-        return llm_positions, mrope_position_delta
+        return torch.from_numpy(llm_positions), mrope_position_delta
```
[the remark this answers is not in the document]
Comment (lgeiger) [reply vllm/model_executor/models/qwen3_vl.py:1491]: The same is true for `torch.cat()`, so this isn't an issue here.

**Review by DarkLight1337 — COMMENTED** [review]
Actually I'm thinking of directly using the positions from `mm_features` instead of having to calculate the mask again, WDYT?

**Comment by DarkLight1337** [comment]
We can pass `req_state.mm_features` which includes `mm_position` (`PlaceholderRange`) into `get_mrope_input_positions`.

**Comment by DarkLight1337** [comment]
Opened https://github.com/vllm-project/vllm/pull/28399

**Review by DarkLight1337 — APPROVED** [review]
Thanks, LGTM

**Comment by DarkLight1337** [comment]
Feel free to work on this yourself!
